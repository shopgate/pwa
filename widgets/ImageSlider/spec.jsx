import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
import ImageSliderWidget from './index';

const mockedStore = configureStore();
/**
 * Creates component
 * @param {Object} props Component props.
 * @return {ReactWrapper}
 */
const createComponent = (props = {}) => mount(
  <Provider store={mockedStore({})}>
    <ImageSliderWidget {...props} />
  </Provider>,
  mockRenderOptions
);

describe('<ImageSliderWidget />', () => {
  const testImage = {
    image: 'http://placehold.it/350x150',
    link: 'http://example.com',
  };

  const testImage2 = {
    image: 'http://placehold.it/10x10',
    link: 'http://other.example.com',
  };

  const testSettings = {
    autostart: false,
    delay: 7000,
    pagination: true,
    loop: false,
    images: [],
  };

  it('should render the slider with the correct number of images', () => {
    const settings = {
      ...testSettings,
      images: [
        testImage,
        testImage2,
      ],
    };

    const wrapper = createComponent({ settings });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('SwiperItem').length).toBe(settings.images.length);

    const images = wrapper.find('img');
    expect(images.length).toBe(settings.images.length);
  });

  it('should map the correct image settings to the components', () => {
    expect.assertions(3);
    const settings = {
      ...testSettings,
      images: [
        testImage,
        testImage2,
      ],
    };

    const wrapper = createComponent({ settings });

    expect(wrapper).toMatchSnapshot();
    const images = wrapper.find('div.swiper-slide img');
    images.forEach((image, index) => {
      const imageProps = image.props();
      const imageSettings = settings.images[index];

      expect(imageProps.src).toBe(imageSettings.image);
    });
  });

  it('should render no slider with just a single image', () => {
    const settings = {
      ...testSettings,
      images: [
        testImage,
      ],
    };

    const wrapper = createComponent({ settings });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('SwiperItem').length).toBe(0);
    expect(wrapper.find('img').length).toBe(1);
  });

  it('should render the images unlinked if no link is set', () => {
    const settings = {
      ...testSettings,
      images: [
        {
          ...testImage,
          link: null,
        },
      ],
    };

    const wrapper = createComponent({ settings });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Link').length).toBe(0);
    expect(wrapper.find('img').length).toBe(1);
  });

  it('should render the images with links', () => {
    const settings = {
      ...testSettings,
      images: [
        {
          // Add an image without a link
          ...testImage,
          link: null,
        },
        testImage2,
      ],
    };

    const wrapper = createComponent({ settings });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Link').length).toBe(settings.images.length - 1);
  });
});
