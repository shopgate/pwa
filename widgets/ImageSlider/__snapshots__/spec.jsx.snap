// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageSliderWidget /> should map the correct image settings to the components 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ImageSliderWidget
    className=""
    settings={
      Object {
        "autostart": false,
        "delay": 7000,
        "images": Array [
          Object {
            "image": "http://placehold.it/350x150",
            "link": "http://example.com",
          },
          Object {
            "image": "http://placehold.it/10x10",
            "link": "http://other.example.com",
          },
        ],
        "loop": false,
        "pagination": true,
      }
    }
  >
    <Swiper
      autoPlay={false}
      className=""
      classNames={Object {}}
      controls={false}
      indicators={true}
      initialSlide={0}
      interval={7000}
      loop={false}
      maxIndicators={null}
      onSlideChange={[Function]}
      rebuildOnUpdate={false}
      slidesPerView={1}
      snapItems={true}
    >
      <div
        className="css-1hlki0z"
      >
        <ReactIdSwiper
          ContainerEl="div"
          WrapperEl="div"
          autoplay={false}
          containerClass="css-rdsogp"
          freeMode={false}
          getSwiper={[Function]}
          initialSlide={0}
          loop={false}
          modules={
            Array [
              Object {
                "create": [Function],
                "name": "pagination",
                "on": Object {
                  "activeIndexChange": [Function],
                  "click": [Function],
                  "destroy": [Function],
                  "init": [Function],
                  "slidesLengthChange": [Function],
                  "snapGridLengthChange": [Function],
                  "snapIndexChange": [Function],
                },
                "params": Object {
                  "pagination": Object {
                    "bulletActiveClass": "swiper-pagination-bullet-active",
                    "bulletClass": "swiper-pagination-bullet",
                    "bulletElement": "span",
                    "clickable": false,
                    "clickableClass": "swiper-pagination-clickable",
                    "currentClass": "swiper-pagination-current",
                    "dynamicBullets": false,
                    "dynamicMainBullets": 1,
                    "el": null,
                    "formatFractionCurrent": [Function],
                    "formatFractionTotal": [Function],
                    "hiddenClass": "swiper-pagination-hidden",
                    "hideOnClick": false,
                    "lockClass": "swiper-pagination-lock",
                    "modifierClass": "swiper-pagination-",
                    "progressbarFillClass": "swiper-pagination-progressbar-fill",
                    "progressbarOpposite": false,
                    "progressbarOppositeClass": "swiper-pagination-progressbar-opposite",
                    "renderBullet": null,
                    "renderCustom": null,
                    "renderFraction": null,
                    "renderProgressbar": null,
                    "totalClass": "swiper-pagination-total",
                    "type": "bullets",
                  },
                },
              },
              Object {
                "create": [Function],
                "name": "navigation",
                "on": Object {
                  "click": [Function],
                  "destroy": [Function],
                  "fromEdge": [Function],
                  "init": [Function],
                  "toEdge": [Function],
                },
                "params": Object {
                  "navigation": Object {
                    "disabledClass": "swiper-button-disabled",
                    "hiddenClass": "swiper-button-hidden",
                    "hideOnClick": false,
                    "lockClass": "swiper-button-lock",
                    "nextEl": null,
                    "prevEl": null,
                  },
                },
              },
            ]
          }
          navigation={
            Object {
              "nextEl": null,
              "prevEl": null,
            }
          }
          on={
            Object {
              "slideChange": [Function],
            }
          }
          pagination={
            Object {
              "bulletActiveClass": "css-hpsgy2",
              "bulletClass": "css-cvsmqf",
              "el": ".swiper-pagination",
              "type": "bullets",
            }
          }
          rebuildOnUpdate={false}
          renderNextButton={[Function]}
          renderPagination={[Function]}
          renderParallax={[Function]}
          renderPrevButton={[Function]}
          renderScrollbar={[Function]}
          slideClass="swiper-slide"
          slidesPerView={1}
          wrapperClass="swiper-wrapper"
        >
          <div
            className="css-rdsogp"
          >
            <div
              className="swiper-wrapper"
            >
              <SwiperItem
                className="swiper-slide"
                key=".$http=2//placehold.it/350x150"
              >
                <div
                  className="swiper-slide"
                  data-test-id="Slider"
                >
                  <Connect(Link)
                    className="css-63oe3q"
                    data-test-id="withLink"
                    href="http://example.com"
                  >
                    <Link
                      className="css-63oe3q"
                      data-test-id="withLink"
                      disabled={false}
                      historyPush={[Function]}
                      historyReplace={[Function]}
                      href="http://example.com"
                      replace={false}
                      state={Object {}}
                      tag="div"
                    >
                      <div
                        className="css-c6ki83 css-63oe3q"
                        data-test-id="link: http://example.com"
                        onClick={[Function]}
                        role="link"
                      >
                        <img
                          className={
                            Object {
                              "data-css-1rozygh": "",
                            }
                          }
                          data-test-id="link : undefined"
                          src="http://placehold.it/350x150"
                        />
                      </div>
                    </Link>
                  </Connect(Link)>
                </div>
              </SwiperItem>
              <SwiperItem
                className="swiper-slide"
                key=".$http=2//placehold.it/10x10"
              >
                <div
                  className="swiper-slide"
                  data-test-id="Slider"
                >
                  <Connect(Link)
                    className="css-63oe3q"
                    data-test-id="withLink"
                    href="http://other.example.com"
                  >
                    <Link
                      className="css-63oe3q"
                      data-test-id="withLink"
                      disabled={false}
                      historyPush={[Function]}
                      historyReplace={[Function]}
                      href="http://other.example.com"
                      replace={false}
                      state={Object {}}
                      tag="div"
                    >
                      <div
                        className="css-c6ki83 css-63oe3q"
                        data-test-id="link: http://other.example.com"
                        onClick={[Function]}
                        role="link"
                      >
                        <img
                          className={
                            Object {
                              "data-css-1rozygh": "",
                            }
                          }
                          data-test-id="link : undefined"
                          src="http://placehold.it/10x10"
                        />
                      </div>
                    </Link>
                  </Connect(Link)>
                </div>
              </SwiperItem>
            </div>
            <div
              className="swiper-pagination"
            />
          </div>
        </ReactIdSwiper>
      </div>
    </Swiper>
  </ImageSliderWidget>
</Provider>
`;

exports[`<ImageSliderWidget /> should render no slider with just a single image 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ImageSliderWidget
    className=""
    settings={
      Object {
        "autostart": false,
        "delay": 7000,
        "images": Array [
          Object {
            "image": "http://placehold.it/350x150",
            "link": "http://example.com",
          },
        ],
        "loop": false,
        "pagination": true,
      }
    }
  >
    <div
      className=""
    >
      <Connect(Link)
        className="css-63oe3q"
        data-test-id="withLink"
        href="http://example.com"
      >
        <Link
          className="css-63oe3q"
          data-test-id="withLink"
          disabled={false}
          historyPush={[Function]}
          historyReplace={[Function]}
          href="http://example.com"
          replace={false}
          state={Object {}}
          tag="div"
        >
          <div
            className="css-c6ki83 css-63oe3q"
            data-test-id="link: http://example.com"
            onClick={[Function]}
            role="link"
          >
            <img
              className={
                Object {
                  "data-css-1rozygh": "",
                }
              }
              data-test-id="link : undefined"
              src="http://placehold.it/350x150"
            />
          </div>
        </Link>
      </Connect(Link)>
    </div>
  </ImageSliderWidget>
</Provider>
`;

exports[`<ImageSliderWidget /> should render the images unlinked if no link is set 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ImageSliderWidget
    className=""
    settings={
      Object {
        "autostart": false,
        "delay": 7000,
        "images": Array [
          Object {
            "image": "http://placehold.it/350x150",
            "link": null,
          },
        ],
        "loop": false,
        "pagination": true,
      }
    }
  >
    <div
      className=""
    >
      <img
        className={
          Object {
            "data-css-1rozygh": "",
          }
        }
        data-test-id="link : undefined"
        src="http://placehold.it/350x150"
      />
    </div>
  </ImageSliderWidget>
</Provider>
`;

exports[`<ImageSliderWidget /> should render the images with links 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ImageSliderWidget
    className=""
    settings={
      Object {
        "autostart": false,
        "delay": 7000,
        "images": Array [
          Object {
            "image": "http://placehold.it/350x150",
            "link": null,
          },
          Object {
            "image": "http://placehold.it/10x10",
            "link": "http://other.example.com",
          },
        ],
        "loop": false,
        "pagination": true,
      }
    }
  >
    <Swiper
      autoPlay={false}
      className=""
      classNames={Object {}}
      controls={false}
      indicators={true}
      initialSlide={0}
      interval={7000}
      loop={false}
      maxIndicators={null}
      onSlideChange={[Function]}
      rebuildOnUpdate={false}
      slidesPerView={1}
      snapItems={true}
    >
      <div
        className="css-1hlki0z"
      >
        <ReactIdSwiper
          ContainerEl="div"
          WrapperEl="div"
          autoplay={false}
          containerClass="css-rdsogp"
          freeMode={false}
          getSwiper={[Function]}
          initialSlide={0}
          loop={false}
          modules={
            Array [
              Object {
                "create": [Function],
                "name": "pagination",
                "on": Object {
                  "activeIndexChange": [Function],
                  "click": [Function],
                  "destroy": [Function],
                  "init": [Function],
                  "slidesLengthChange": [Function],
                  "snapGridLengthChange": [Function],
                  "snapIndexChange": [Function],
                },
                "params": Object {
                  "pagination": Object {
                    "bulletActiveClass": "swiper-pagination-bullet-active",
                    "bulletClass": "swiper-pagination-bullet",
                    "bulletElement": "span",
                    "clickable": false,
                    "clickableClass": "swiper-pagination-clickable",
                    "currentClass": "swiper-pagination-current",
                    "dynamicBullets": false,
                    "dynamicMainBullets": 1,
                    "el": null,
                    "formatFractionCurrent": [Function],
                    "formatFractionTotal": [Function],
                    "hiddenClass": "swiper-pagination-hidden",
                    "hideOnClick": false,
                    "lockClass": "swiper-pagination-lock",
                    "modifierClass": "swiper-pagination-",
                    "progressbarFillClass": "swiper-pagination-progressbar-fill",
                    "progressbarOpposite": false,
                    "progressbarOppositeClass": "swiper-pagination-progressbar-opposite",
                    "renderBullet": null,
                    "renderCustom": null,
                    "renderFraction": null,
                    "renderProgressbar": null,
                    "totalClass": "swiper-pagination-total",
                    "type": "bullets",
                  },
                },
              },
              Object {
                "create": [Function],
                "name": "navigation",
                "on": Object {
                  "click": [Function],
                  "destroy": [Function],
                  "fromEdge": [Function],
                  "init": [Function],
                  "toEdge": [Function],
                },
                "params": Object {
                  "navigation": Object {
                    "disabledClass": "swiper-button-disabled",
                    "hiddenClass": "swiper-button-hidden",
                    "hideOnClick": false,
                    "lockClass": "swiper-button-lock",
                    "nextEl": null,
                    "prevEl": null,
                  },
                },
              },
            ]
          }
          navigation={
            Object {
              "nextEl": null,
              "prevEl": null,
            }
          }
          on={
            Object {
              "slideChange": [Function],
            }
          }
          pagination={
            Object {
              "bulletActiveClass": "css-hpsgy2",
              "bulletClass": "css-cvsmqf",
              "el": ".swiper-pagination",
              "type": "bullets",
            }
          }
          rebuildOnUpdate={false}
          renderNextButton={[Function]}
          renderPagination={[Function]}
          renderParallax={[Function]}
          renderPrevButton={[Function]}
          renderScrollbar={[Function]}
          slideClass="swiper-slide"
          slidesPerView={1}
          wrapperClass="swiper-wrapper"
        >
          <div
            className="css-rdsogp"
          >
            <div
              className="swiper-wrapper"
            >
              <SwiperItem
                className="swiper-slide"
                key=".$http=2//placehold.it/350x150"
              >
                <div
                  className="swiper-slide"
                  data-test-id="Slider"
                >
                  <img
                    className={
                      Object {
                        "data-css-1rozygh": "",
                      }
                    }
                    data-test-id="link : undefined"
                    src="http://placehold.it/350x150"
                  />
                </div>
              </SwiperItem>
              <SwiperItem
                className="swiper-slide"
                key=".$http=2//placehold.it/10x10"
              >
                <div
                  className="swiper-slide"
                  data-test-id="Slider"
                >
                  <Connect(Link)
                    className="css-63oe3q"
                    data-test-id="withLink"
                    href="http://other.example.com"
                  >
                    <Link
                      className="css-63oe3q"
                      data-test-id="withLink"
                      disabled={false}
                      historyPush={[Function]}
                      historyReplace={[Function]}
                      href="http://other.example.com"
                      replace={false}
                      state={Object {}}
                      tag="div"
                    >
                      <div
                        className="css-c6ki83 css-63oe3q"
                        data-test-id="link: http://other.example.com"
                        onClick={[Function]}
                        role="link"
                      >
                        <img
                          className={
                            Object {
                              "data-css-1rozygh": "",
                            }
                          }
                          data-test-id="link : undefined"
                          src="http://placehold.it/10x10"
                        />
                      </div>
                    </Link>
                  </Connect(Link)>
                </div>
              </SwiperItem>
            </div>
            <div
              className="swiper-pagination"
            />
          </div>
        </ReactIdSwiper>
      </div>
    </Swiper>
  </ImageSliderWidget>
</Provider>
`;

exports[`<ImageSliderWidget /> should render the slider with the correct number of images 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ImageSliderWidget
    className=""
    settings={
      Object {
        "autostart": false,
        "delay": 7000,
        "images": Array [
          Object {
            "image": "http://placehold.it/350x150",
            "link": "http://example.com",
          },
          Object {
            "image": "http://placehold.it/10x10",
            "link": "http://other.example.com",
          },
        ],
        "loop": false,
        "pagination": true,
      }
    }
  >
    <Swiper
      autoPlay={false}
      className=""
      classNames={Object {}}
      controls={false}
      indicators={true}
      initialSlide={0}
      interval={7000}
      loop={false}
      maxIndicators={null}
      onSlideChange={[Function]}
      rebuildOnUpdate={false}
      slidesPerView={1}
      snapItems={true}
    >
      <div
        className="css-1hlki0z"
      >
        <ReactIdSwiper
          ContainerEl="div"
          WrapperEl="div"
          autoplay={false}
          containerClass="css-rdsogp"
          freeMode={false}
          getSwiper={[Function]}
          initialSlide={0}
          loop={false}
          modules={
            Array [
              Object {
                "create": [Function],
                "name": "pagination",
                "on": Object {
                  "activeIndexChange": [Function],
                  "click": [Function],
                  "destroy": [Function],
                  "init": [Function],
                  "slidesLengthChange": [Function],
                  "snapGridLengthChange": [Function],
                  "snapIndexChange": [Function],
                },
                "params": Object {
                  "pagination": Object {
                    "bulletActiveClass": "swiper-pagination-bullet-active",
                    "bulletClass": "swiper-pagination-bullet",
                    "bulletElement": "span",
                    "clickable": false,
                    "clickableClass": "swiper-pagination-clickable",
                    "currentClass": "swiper-pagination-current",
                    "dynamicBullets": false,
                    "dynamicMainBullets": 1,
                    "el": null,
                    "formatFractionCurrent": [Function],
                    "formatFractionTotal": [Function],
                    "hiddenClass": "swiper-pagination-hidden",
                    "hideOnClick": false,
                    "lockClass": "swiper-pagination-lock",
                    "modifierClass": "swiper-pagination-",
                    "progressbarFillClass": "swiper-pagination-progressbar-fill",
                    "progressbarOpposite": false,
                    "progressbarOppositeClass": "swiper-pagination-progressbar-opposite",
                    "renderBullet": null,
                    "renderCustom": null,
                    "renderFraction": null,
                    "renderProgressbar": null,
                    "totalClass": "swiper-pagination-total",
                    "type": "bullets",
                  },
                },
              },
              Object {
                "create": [Function],
                "name": "navigation",
                "on": Object {
                  "click": [Function],
                  "destroy": [Function],
                  "fromEdge": [Function],
                  "init": [Function],
                  "toEdge": [Function],
                },
                "params": Object {
                  "navigation": Object {
                    "disabledClass": "swiper-button-disabled",
                    "hiddenClass": "swiper-button-hidden",
                    "hideOnClick": false,
                    "lockClass": "swiper-button-lock",
                    "nextEl": null,
                    "prevEl": null,
                  },
                },
              },
            ]
          }
          navigation={
            Object {
              "nextEl": null,
              "prevEl": null,
            }
          }
          on={
            Object {
              "slideChange": [Function],
            }
          }
          pagination={
            Object {
              "bulletActiveClass": "css-hpsgy2",
              "bulletClass": "css-cvsmqf",
              "el": ".swiper-pagination",
              "type": "bullets",
            }
          }
          rebuildOnUpdate={false}
          renderNextButton={[Function]}
          renderPagination={[Function]}
          renderParallax={[Function]}
          renderPrevButton={[Function]}
          renderScrollbar={[Function]}
          slideClass="swiper-slide"
          slidesPerView={1}
          wrapperClass="swiper-wrapper"
        >
          <div
            className="css-rdsogp"
          >
            <div
              className="swiper-wrapper"
            >
              <SwiperItem
                className="swiper-slide"
                key=".$http=2//placehold.it/350x150"
              >
                <div
                  className="swiper-slide"
                  data-test-id="Slider"
                >
                  <Connect(Link)
                    className="css-63oe3q"
                    data-test-id="withLink"
                    href="http://example.com"
                  >
                    <Link
                      className="css-63oe3q"
                      data-test-id="withLink"
                      disabled={false}
                      historyPush={[Function]}
                      historyReplace={[Function]}
                      href="http://example.com"
                      replace={false}
                      state={Object {}}
                      tag="div"
                    >
                      <div
                        className="css-c6ki83 css-63oe3q"
                        data-test-id="link: http://example.com"
                        onClick={[Function]}
                        role="link"
                      >
                        <img
                          className={
                            Object {
                              "data-css-1rozygh": "",
                            }
                          }
                          data-test-id="link : undefined"
                          src="http://placehold.it/350x150"
                        />
                      </div>
                    </Link>
                  </Connect(Link)>
                </div>
              </SwiperItem>
              <SwiperItem
                className="swiper-slide"
                key=".$http=2//placehold.it/10x10"
              >
                <div
                  className="swiper-slide"
                  data-test-id="Slider"
                >
                  <Connect(Link)
                    className="css-63oe3q"
                    data-test-id="withLink"
                    href="http://other.example.com"
                  >
                    <Link
                      className="css-63oe3q"
                      data-test-id="withLink"
                      disabled={false}
                      historyPush={[Function]}
                      historyReplace={[Function]}
                      href="http://other.example.com"
                      replace={false}
                      state={Object {}}
                      tag="div"
                    >
                      <div
                        className="css-c6ki83 css-63oe3q"
                        data-test-id="link: http://other.example.com"
                        onClick={[Function]}
                        role="link"
                      >
                        <img
                          className={
                            Object {
                              "data-css-1rozygh": "",
                            }
                          }
                          data-test-id="link : undefined"
                          src="http://placehold.it/10x10"
                        />
                      </div>
                    </Link>
                  </Connect(Link)>
                </div>
              </SwiperItem>
            </div>
            <div
              className="swiper-pagination"
            />
          </div>
        </ReactIdSwiper>
      </div>
    </Swiper>
  </ImageSliderWidget>
</Provider>
`;
